argilla/tests/integration/test_export_dataset.py

```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
 
-    assert new_dataset.settings.fields[0].name == "text"
-    assert new_dataset.settings.questions[0].name == "label"
+@pytest.fixture
+def token():
+    return os.getenv("HF_TOKEN_ARGILLA_INTERNAL_TESTING") or os.getenv("HF_TOKEN")
+
+
+@pytest.mark.parametrize("with_records", [True, False])
+class TestDiskImportExportMixin:
+    def test_export_dataset_to_disk(self, dataset: rg.Dataset, mock_data: List[dict[str, Any]], with_records: bool):
+        dataset.records.log(records=mock_data)
+
+        with TemporaryDirectory() as temp_dir:
+            output_dir = dataset.to_disk(path=temp_dir, with_records=with_records)
+
+            records_path = os.path.join(output_dir, rg.Dataset._default_records_path)
+            if with_records:
+                assert os.path.exists(records_path)
+                with open(records_path, "r") as f:
+                    exported_records = json.load(f)
+            else:
+                assert not os.path.exists(records_path)
+
+            settings_path = os.path.join(output_dir, rg.Dataset._default_settings_path)
+            assert os.path.exists(settings_path)
+            with open(settings_path, "r") as f:
+                exported_settings = json.load(f)
+
+            dataset_path = os.path.join(output_dir, rg.Dataset._default_dataset_path)
+            assert os.path.exists(dataset_path)
+            with open(dataset_path, "r") as f:
+                exported_dataset = json.load(f)
+
+        if with_records:
+            assert len(exported_records) == len(mock_data)
+            assert exported_records[0]["fields"]["text"] == "Hello World, how are you?"
+            assert exported_records[0]["suggestions"]["label"]["value"] == "positive"
+
+            assert exported_settings["fields"][0]["name"] == "text"
+            assert exported_settings["questions"][0]["name"] == "label"
+
+            assert exported_dataset["name"] == dataset.name
+
+    def test_import_dataset_from_disk(
+        self, dataset: rg.Dataset, client, mock_data: List[dict[str, Any]], with_records: bool
+    ):
+        dataset.records.log(records=mock_data)
+
+        with TemporaryDirectory() as temp_dir:
+            output_dir = dataset.to_disk(path=temp_dir, with_records=with_records)
+            new_dataset = rg.Dataset.from_disk(output_dir, client=client, with_records=with_records)
+
+        if with_records:
+            for i, record in enumerate(new_dataset.records(with_suggestions=True)):
+                assert record.fields["text"] == mock_data[i]["text"]
+                assert record.suggestions["label"].value == mock_data[i]["label"]
+        else:
+            assert len(new_dataset.records.to_list()) == 0
+
+        assert new_dataset.settings.fields[0].name == "text"
+        assert new_dataset.settings.questions[0].name == "label"
+
+
+@pytest.mark.skipif(
+    not (os.getenv("HF_TOKEN_ARGILLA_INTERNAL_TESTING") or os.getenv("HF_TOKEN")),
```

Comment:
Maybe is simpler using the standard naming.
```suggestion
    not os.getenv("HF_TOKEN"),
```